packages/kbn-lens-embeddable-utils/config_builder/types.ts

```diff
@@ -39,7 +39,7 @@ export interface TimeRange {
   type: 'relative' | 'absolute';
 }
 
-export type LensLayerQuery = string;
```

Comment:
can you please explain why is this change necesarry ? thanks!

Reply:
We need to be able to pass more than the formula, but also the format and a custom label. e.g:

https://github.com/elastic/kibana/pull/174298/files#diff-74e19f809f3a75344f6e4a608cf65fa11a08075065cb442582e983db0399f5ab

Reply:
label and format should not be set on the "query", in fact they are already properties of a BaseLayer:
https://github.com/elastic/kibana/pull/174350/files#diff-2ad458414d2865475a59633d7a2651d6bde8a4cad566608f66327aefef36c5c3R63
query should be limited to just what we need to fetch the data.

Reply:
also, it got me thinking, right now we can set label and format for the layer, but if layer has more values (like in metric we have metric and secondary metric ... would we ever want to set some of those properties different for each of them ? like separate format or label ? not sure how much of this is actually supported in lens @stratoula 
 but if we want to add that then currently we have an issue ...

would something like this work for you:

```
{ ...
  metric: LensValueConfig;
  secondaryMetric: LensValueConfig;
  maxValue: LensValueConfig;
  ...
```

```
interface LensValueConfig {
   query: LensLayerQuery;
   format: ...
   label: string;
   lineThickness?: number;
  color?: string;
}
```

this way we keep query simple, only containing what is needed for data fetching, but we still allow to set other properties per value and keeps the types clean.

Reply:
For the primary and secondary value yes we support different label and format so this makes sense to me

Reply:
for now lets introduce the minimal set of required features, if we need to be able to define label and format for some of the values lets introduce that as described above.

Reply:
I agree ++